platforms/core-runtime/base-services/src/main/java/org/gradle/internal/os/OperatingSystem.java

```diff
@@ -428,4 +431,17 @@ protected String getArch() {
             return super.getArch();
         }
     }
+
+    static class Aix extends Unix {
```

Comment:
I am confused by this error
```
java.lang.AssertionError: Architecture Violation [Priority: MEDIUM] - Rule 'classes that are in Gradle internal API packages and classes written in Java or Groovy and not synthetic classes should be annotated (directly or via its package) with @org.gradle.api.NonNullApi' was violated (1 times):
Class <org.gradle.internal.os.OperatingSystem$Aix> is not annotated (directly or via its package) with @org.gradle.api.NonNullApi in (OperatingSystem.java:0)
```

Other inner classes here also override `getFamilyName` and `getOsPrefix`, but don't produce any errors.

Reply:
Yeah this check does not work properly for nested classes. I'd remove this annotation here and annotate the whole package ([example](https://github.com/gradle/gradle/blob/b466244a613f6649df17e32a323f1fe43dad4d2c/subprojects/core-api/src/main/java/org/gradle/package-info.java#L20-L21)).

Then, you will need to run `./gradlew architecture-test:test -ParchunitRefreeze` and commit the changes to the nullability file.